src/app/src/pages/eval/components/EvalOutputPromptDialog.tsx

```diff
@@ -395,6 +397,25 @@ export default function EvalOutputPromptDialog({
       ],
     );
 
+  const formatMessagesForCopy = useCallback((messages: Message[]): string => {
+    return messages
+      .filter((msg): msg is LoadedMessage => !msg.loading && msg.content !== null)
+      .map((msg) => {
+        const roleLabel =
+          msg.role === 'user' ? 'User' : msg.role === 'assistant' ? 'Assistant' : 'System';
+        return `[${roleLabel}]\n${msg.content}`;
+      })
+      .join('\n\n');
+  }, []);
+
+  const copyMessagesToClipboard = useCallback(async () => {
```

Comment:
this copies _all_ messages? I would expect it to only copy the given message it's on?